includes/admin/stripe-settings.php

```diff
@@ -15,14 +15,14 @@
 		],
 		'title'                               => [
 			'title'       => __( 'Title', 'woocommerce-gateway-stripe' ),
-			'type'        => 'text',
+			'type'        => 'safe_text',
```

Comment:
Does `safe_text` makes a difference with `text`? I've tested both and haven't found any.

Reply:
Yes, it does. In WooCommerce v6.6, `safe_text` has a separate validation function than `text`. The validation function call can be seen in [class-wc-rest-stripe-settings-controller.php::validate_field( ... )](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2362/commits/fb7c7b0374494176c4b862af8a51823c0df2c937#diff-679186f0c3e522393b5394f6e65890f1b2bff5ef64efb250eee90d6348dc841eR365):
```php
return $this->gateway->{'validate_' . $field_type . '_field'}( $field_key, $field_value );
```

Reply:
So what happens in WC < 6.6? Is `safe_text` even recognized? Do we need to make this a `WC_VERSION < 6.6 ? 'text' : 'safe_text'` type thing? 🤔

Reply:
I seems it'll default to `text` https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2362/files#diff-679186f0c3e522393b5394f6e65890f1b2bff5ef64efb250eee90d6348dc841eR366

Reply:
What I understand is that if the user upgrades WCStripe the issue will still exist until the form is saved again. But if the user stays with their WCStripe version and update WC to 6.6+ the fix will be applied at the moment the title is displayed.

It would be nice to fix for the case the user upgrades WCStripe only and doesn't save/update the form yet.

Reply:
> So what happens in WC < 6.6? Is `safe_text` even recognized? Do we need to make this a `WC_VERSION < 6.6 ? 'text' : 'safe_text'` type thing? 🤔

That's correct, `safe_text` is not recognized in WC < 6.6. But that's already been handled in this PR, like follows:
```php
private function validate_field( $field_key, $field_value, $form_fields ) {
  ...
  if ( key_exists( $field_key, $form_fields ) ) {
    $field_type = $form_fields[ $field_key ]['type'];
    // Checks if `validate_safe_text_field` exists.
    if ( is_callable( [ $this->gateway, 'validate_' . $field_type . '_field' ] ) ) {
      return $this->gateway->{'validate_' . $field_type . '_field'}( $field_key, $field_value );
    }
  }
  // Otherwise, defaults to `validate_text_field`.
  return $this->gateway->validate_text_field( $field_key, $field_value );
}
```
In other words, we do not need to apply the suggested ternary since `validate_field( ... )` defaults to `validate_text_field( ... )` already. [See code-block for more details](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2362/commits/fb7c7b0374494176c4b862af8a51823c0df2c937#diff-679186f0c3e522393b5394f6e65890f1b2bff5ef64efb250eee90d6348dc841eR364-R366).

> I thought this was called regardless of what the type of the field is? I think the safe_text is only detected by WC >= 6.6 where those versions of WC will apply sanitation in addition to whatever validation/sanitation we do. I doubt they've updated older versions of WC to handle the safe_text type.

Correct, they have not updated older versions of WC to include `safe_text` but as stated above, in that event, this PR defaults to `validate_text_field( ... )`. In other words, it treats it as if the type was `text`.

> Have we confirmed that WC < 6.6 doesn't just skip the WC core validation/sanitation entirely if we use safe_text?

Yes, we can confirm that if we use `safe_text` as the type, it will not skip validation. This can be seen in `validate_field( ... )`: it will default to `validate_text_field( ... )`.

Reply:
> What I understand is that if the user upgrades WCStripe the issue will still exist until the form is saved again. But if the user stays with their WCStripe version and update WC to 6.6+ the fix will be applied at the moment the title is displayed.
> 
> It would be nice to fix for the case the user upgrades WCStripe only and doesn't save/update the form yet.

Good call. We could do something similar to what WC is doing and apply the validation at the source of the output. In other words, we could validate at [class-wc-rest-stripe-settings-controller.php::get_settings()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/admin/class-wc-rest-stripe-settings-controller.php#L237-L271). Thoughts?

Reply:
> We could do something similar to what WC is doing and apply the validation at the source of the output. In other words, we could validate at [class-wc-rest-stripe-settings-controller.php::get_settings()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/admin/class-wc-rest-stripe-settings-controller.php#L237-L271). Thoughts?

I think applying the sanitation at the source of the output is a good idea and seems to be a general WP development recommendation. That is, of course, unless we're using this all over the place 😛 . In that case it'd probably be best to try to sanitize just once ☺️